cloudinit/distros/__init__.py

```diff
@@ -998,6 +1001,26 @@ def do_as(self, command: list, user: str, cwd: str = "", **kwargs):
             **kwargs,
         )
 
+    @staticmethod
+    def build_dhclient_cmd(
+        path: str,
+        lease_file: str,
+        pid_file: str,
+        interface: str,
+        config_file: str,
+    ) -> list:
+        return [
+            path,
+            "-1",
+            "-v",
+            "-lf",
+            lease_file,
+            "-pf",
+            pid_file,
+            "-sf",
+            "/bin/true",
+        ] + (["-cf", config_file, interface] if config_file else [interface])
+
```

Comment:
just a personal preference, but for readability, I think I'd rather see the args built up over a few lines and conditionals than trying to get it all into one statement. e.g.

```
args = [
            path,
            "-1",
            "-v",
            "-lf",
            lease_file,
            "-pf",
            pid_file,
            "-sf",
            "/bin/true",
        ]

if config_file:
    args.extend(["-cf", config_file, interface])

args.append(interface)
```

I may be alone in this preference, so feel free to ignore :D

Reply:
i like the idea!
but this, too is @holmanb's code.